server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -128,14 +128,17 @@ func (s *BuildBuddyServer) GetGroup(ctx context.Context, req *grpb.GetGroupReque
 	if userDB == nil {
 		return nil, status.UnimplementedError("Not Implemented")
 	}
-	group := &tables.Group{}
 	urlIdentifier := strings.TrimSpace(req.GetUrlIdentifier())
-	if urlIdentifier != "" {
-		group.URLIdentifier = &urlIdentifier
+	if urlIdentifier == "" {
+		return nil, status.InvalidArgumentError("URL identifier is required.")
 	}
-	if err := userDB.FillGroup(ctx, group); err != nil {
+	group, err := userDB.GetGroupByURLIdentifier(ctx, urlIdentifier)
```

Comment:
It's a little atypical in go to return a nil value if err is nil, but it's not a rule.

You might consolidate this and just return status.NotFoundError("The requested organization was not found.") from inside GetGroupByUrlIdentifier. and then you don't need the if group == nil check here.

Reply:
I don't want `GetGroupByURLIdentifier` to return an error in the case where we want to check whether an URL identifier is already taken -- in cases where it's not already taken, `nil` is returned and that's not really an error. I'm sort of using `nil` like an "Optional" value that is not present.

Reply:
Updated to check errors per our offline discussion